android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/CustomTabConfig.kt

```diff
@@ -9,34 +9,41 @@ import android.graphics.Bitmap
 import android.os.Bundle
 import androidx.annotation.ColorInt
 import androidx.browser.customtabs.CustomTabsIntent
+import androidx.browser.customtabs.CustomTabsIntent.COLOR_SCHEME_DARK
+import androidx.browser.customtabs.CustomTabsIntent.COLOR_SCHEME_LIGHT
+import androidx.browser.customtabs.CustomTabsIntent.ColorScheme
 import androidx.browser.customtabs.CustomTabsSessionToken
 
 /**
  * Holds configuration data for a Custom Tab.
  *
- * @property toolbarColor Background color for the toolbar.
- * @property closeButtonIcon Custom icon of the back button on the toolbar.
+ * @property colorScheme Optional [ColorScheme] to apply to the Custom Tab UI.
+ * @property defaultColorSchemeParams Optional default [ColorSchemeParams].
+ * @property lightColorSchemeParams Optional [ColorSchemeParams] for [COLOR_SCHEME_LIGHT].
+ * @property darkColorSchemeParams Optional [ColorSchemeParams] for [COLOR_SCHEME_DARK].
+ * @property closeButtonIcon Optional custom icon of the back button on the toolbar.
  * @property enableUrlbarHiding Enables the toolbar to hide as the user scrolls down on the page.
- * @property actionButtonConfig Custom action button on the toolbar.
+ * @property actionButtonConfig Optional custom action button on the toolbar.
  * @property showCloseButton Specifies whether the close button will be shown on the toolbar.
  * @property showShareMenuItem Specifies whether a default share button will be shown in the menu.
  * @property menuItems Custom overflow menu items.
- * @property exitAnimations Bundle containing custom exit animations for the tab.
- * @property navigationBarColor Background color for the navigation bar.
+ * @property exitAnimations Optional [Bundle] containing custom exit animations for the tab.
  * @property titleVisible Whether the title should be shown in the custom tab.
- * @property sessionToken The token associated with the custom tab.
+ * @property sessionToken Optional token associated with the custom tab.
  * @property externalAppType How this custom tab is being displayed.
  */
 data class CustomTabConfig(
-    @ColorInt val toolbarColor: Int? = null,
+    @ColorScheme val colorScheme: Int? = null,
+    val defaultColorSchemeParams: ColorSchemeParams? = null,
+    val lightColorSchemeParams: ColorSchemeParams? = null,
+    val darkColorSchemeParams: ColorSchemeParams? = null,
```

Comment:
Maybe we could have the parameters as part of data classes ? Something like a theme class or similar just to group all of the as parameter so we don't group the list parameters for creating a `CustomTabConfig`, what do you think?

Reply:
Similar to what we did for `ColorSchemeParams`.

Reply:
something like this?
```kotlin
data class CustomTabThemes(
    val defaultColorSchemeParams: ColorSchemeParams? = null,
    val lightColorSchemeParams: ColorSchemeParams? = null,
    val darkColorSchemeParams: ColorSchemeParams? = null,
)
```

Reply:
Yes